internal/worker/controlleragentconfig/worker.go

```diff
@@ -107,6 +133,21 @@ func newWorker(cfg WorkerConfig, internalStates chan string) (*configWorker, err
 	return w, nil
 }
 
+func (w *configWorker) registerHandlers(r *mux.Router) {
+	r.HandleFunc("/reload", w.reloadHandler).
+		Methods(http.MethodGet)
+}
+
+// reloadHandler sends a signal to the configWorker when a config reload is
+// requested.
+func (w *configWorker) reloadHandler(resp http.ResponseWriter, req *http.Request) {
+	select {
+	case <-w.catacomb.Dying():
+	case <-req.Context().Done():
+	case w.reloadRequested <- struct{}{}:
+	}
```

Comment:
Let's try and be good citizens and return a good status code.

```suggestion
	select {
	case <-w.catacomb.Dying():
		resp.WriteHeader(http.StatusInternalServerError)
	case <-req.Context().Done():
		resp.WriteHeader(http.StatusInternalServerError)
	case w.reloadRequested <- struct{}{}:
		resp.WriteHeader(http.StatusNoContent)
	}
```

Reply:
Added in the status requests.